packages/vue-components/src/panels/PanelBase.js

```diff
@@ -94,6 +94,9 @@ export default {
     hasSrc() {
       return this.src && this.src.length > 0;
     },
+    getSrcWithoutFragment() {
+      return this.src.split('#')[0];
```

Comment:
what's this for?

Reply:
and one more bug:

reproduction
- `<panel src=".....formattingContents.md" />`
- expand it

![Untitled](https://user-images.githubusercontent.com/3306138/116177438-5cb75480-a746-11eb-9be1-514af1d9ecb1.png)

Reply:
> what's this for?

Initially we had this piece of code which is mutating the prop `this.src`. 

```
      const hash = getFragmentByHash(this.src);
      if (hash) {
        this.fragment = hash;
        // eslint-disable-next-line prefer-destructuring
        this.src = this.src.split('#')[0];
      }
```

Instead of re-assigning the portion of `src` without the fragment to `this.src` thus mutating it, I decided to turn it into a computed property. On second thought, would it be better if I create another data variable for it instead? 

> and one more bug:
> 
> reproduction
> 
> * `<panel src=".....formattingContents.md" />`
> * expand it
> 
> ![Untitled](https://user-images.githubusercontent.com/3306138/116177438-5cb75480-a746-11eb-9be1-514af1d9ecb1.png)

Thanks for the spot! This happened in `Retriever` where a new Vue instance is instantiated, and the data / method were not passed down to it. I have rectified this accordingly :-)

Reply:
> Initially we had this piece of code which is mutating the prop this.src.

Thanks for clarifying!

> would it be better if I create another data variable for it instead?

No need, `data` should be for things that do change in a component's lifetime (`src` dosen't), a computed property would be more suitable here.